src/components/Calculator.js

```diff
@@ -1,52 +1,41 @@
-import React from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 
-export default class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: 0,
-      next: null,
-      operation: null,
-    };
-    this.handler = this.handler.bind(this);
-  }
+export default function Calculator() {
+  const [state, setState] = useState({ total: 0, next: null, operation: null });
```

Comment:
- [ ] [OPTIONAL] good job refactoring the component to a functional component, I would advise however to use ES6 syntax which would make this an arrow function instead of using the function keyword, this will make the component look something like below:
```
const Calculator = () => {
  return (
    <div>Calculator</div>
  )
}

export default Calculator
```